lib/charms/opensearch/v0/opensearch_locking.py

```diff
@@ -343,7 +344,12 @@ def release(self):
         if host or alt_hosts:
             logger.debug("[Node lock] Checking which unit has opensearch lock")
             # Check if this unit currently has lock
-            if self._unit_with_lock(host) == self._charm.unit.name:
+            # or if there is a stale lock from a unit no longer existing
+            if (
+                self._unit_with_lock(host) == self._charm.unit.name
+                or self._unit_with_lock(host)
+                not in self._charm.model.get_relation(self._ENDPOINT_NAME).units
```

Comment:
Similarly, we can directly use `PeerRelationName` instead of `self._ENDPOINT_NAME`